From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: qeth: avoid recovery during device online setting
References: bnc#583296,LTC#61167
Patch-mainline: Upstream submission made

Symptom:     kernel panic while qeth recovery is running
Problem:     If a qeth device is set online, several initialisation steps
             are performed. If a failure in one of these steps occurs,
             the qeth device is reset into DOWN state. If due to the
             failure a qeth recovery is scheduled and started in another
             thread, this might cause all kinds of conflicts, even a
             kernel panic.
Solution:    Forbid scheduling of a qeth recovery while online processing
             is performed till the card is in state SOFTSETUP.

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/net/qeth_l2_main.c |    1 -
 drivers/s390/net/qeth_l3_main.c |    2 --
 2 files changed, 3 deletions(-)

--- a/drivers/s390/net/qeth_l2_main.c
+++ b/drivers/s390/net/qeth_l2_main.c
@@ -939,7 +939,6 @@ static int __qeth_l2_set_online(struct c
 	QETH_DBF_TEXT(SETUP, 2, "setonlin");
 	QETH_DBF_HEX(SETUP, 2, &card, sizeof(void *));
 
-	qeth_set_allowed_threads(card, QETH_RECOVER_THREAD, 1);
 	recover_flag = card->state;
 	rc = ccw_device_set_online(CARD_RDEV(card));
 	if (rc) {
--- a/drivers/s390/net/qeth_l3_main.c
+++ b/drivers/s390/net/qeth_l3_main.c
@@ -3322,8 +3322,6 @@ static int __qeth_l3_set_online(struct c
 	QETH_DBF_TEXT(SETUP, 2, "setonlin");
 	QETH_DBF_HEX(SETUP, 2, &card, sizeof(void *));
 
-	qeth_set_allowed_threads(card, QETH_RECOVER_THREAD, 1);
-
 	recover_flag = card->state;
 	rc = ccw_device_set_online(CARD_RDEV(card));
 	if (rc) {
